src/petals/client/remote_forward_backward.py

```diff
@@ -84,26 +86,22 @@ async def run_remote_forward(
     kwargs = {key: kwargs[key] for key in rpc_info["keyword_names"]}
 
     # Note: we put keyword arguments in the same order as on a server to prevent f(a=1, b=2) != f(b=2, a=1) errors
-    forward_inputs = (inputs, kwargs)
-
-    # Modify forward_schema to support prompts
+    forward_inputs = nested_flatten(inputs, kwargs)
     args_schema, kwargs_schema = rpc_info["forward_schema"]
-    # TODO: rm this assert when support arbitrary number of input tensors
-    assert len(args_schema) == 1 and len(inputs) == 2
-    forward_schema_with_prompts = (tuple(args_schema * len(inputs)), kwargs_schema)
-
-    if not nested_compare(forward_inputs, forward_schema_with_prompts):
-        raise TypeError(f"Inputs do not match expert input schema. Did you pass the right number of parameters?")
-
-    forward_inputs = nested_flatten(forward_inputs)
+    compression = args_schema[0].compression
+    forward_schema = tuple(
+        BatchTensorDescriptor.from_tensor(arg, compression)
+        for arg in forward_inputs
+    )
     inputs = tuple(tensor.cpu().detach() for tensor in forward_inputs)
+    assert len(inputs) >= len(args_schema) + 1, "Inputs and prompt tensors are necessary for a forward step"
```

Comment:
```suggestion
    assert len(inputs) > len(args_schema), "Inputs and prompt tensors are necessary for a forward step"
```

Reply:
If we assert that we need a specific number of arguments (and specific kind of arguments), maybe we can check the inputs more explicitly?

Reply:
Maybe in the future. Currently, I want to have a better way to provide tensors from clients into servers. I don't want to enhance args_scheme in this PR, but it will be awesome todo.

I will leave a TODO about it here.

Reply:
@mryab I think it's more clear the way @artek0chumak wrote it - so that no one misses the `>` instead of `>=`, and everyone sees that the min number of args is `len(args_schema) + 1`.